### DISCUSSION TITLE
Consumable Items and "Destroy when Empty"

### DISCUSSION THREAD

**Discussion by TheGreatRedDuke** [wildj79/foundryvtt-starfinder#1259] [Q&A]
Firstly, thanks to the devs for their continued hard work on this system. I always look forward to each update and the new features and additions they make. Never sure where the best place to let them know this is, but you have my thanks! Now a question:

---

Hoping some one can point me in the right direction in how to configure things...

When you have a consumable in your inventory, lets say an "Antitoxin, Tier 1" for example, and you use the item, it always seems to remain in the player's inventory, but just with a 0 quantity. Is there a way to configure the item to delete itself when the quantity hits zero?

I've spent a lot of time testing as many different configurations as I can, but can't seem to make this happen.

---

Looking again at Antitoxin, Tier 1 as an example:

You can set the Activation Cost to 1 Standard Action (which is what is should be anyway I believe). Having a Activation Cost on the item presents you with further configuration, including the "Limited Uses" inputs as well as two checkboxes for "Consume on use" and "Destroy when empty".

No configuration combination I can find actually deletes the item when it's used and it's quantity hits 0 which is the behavior I expect from the "Destroy when empty" checkbox.

I've tried setting limited uses (or specifically leaving it blank) and different combinations of all these settings without any luck. If the consumable has a quantity higher than 1, these settings _do_ ensure that the quantity is reduced as expected, but once it hits zero, it the item remains.

---

I'm not sure if this is the expected behavior as designed, if I'm setting it up wrong, or if maybe this is a bug that I should open an issue for. Grenades are another item that have a very similar behavior of remaining in your inventory after use and reaching quantity of 0, but I think that's a separate discussion.

Hopefully someone has some thoughts or insights than can point me in the right direction.

Thanks!

**Comment by ian612** [comment]
Hi there, sorry these discussion posts don't get checked very often but I wanted to provide an answer for you. If this isn't functioning it's likely a bug, since the code is in the system to delete the item automatically when out. If you file an Issue (bug report) then it can get tracked and fixed more easily.
